src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -522,15 +522,10 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
             if (result == null) {
                 // non-protocol related error
-                String errMessage = null;
-                byte[] message = response.getBody();
-                if (message != null) {
-                    errMessage = new String(message);
-                } else {
-                    errMessage = "Status code:" + String.valueOf(response.getStatusCode());
-                }
+                final String errMessage = (response.getBody() == null || response.getBody().length == 0) ? 
+                        "Status code:" + String.valueOf(response.getStatusCode()) : new String(response.getBody());
 
-                Logger.v(TAG, "Server error message:" + errMessage);
+                Logger.e(TAG, "Server error message", errMessage, ADALError.SERVER_ERROR);
                 if (response.getResponseException() != null) {
```

Comment:
why don't we throw anything in case if we don't have Exception in response, but result is null?